ci-operator/step-registry/baremetalds/e2e/ovn/bgp/pre/baremetalds-e2e-ovn-bgp-pre-commands.sh

```diff
@@ -27,119 +50,184 @@ clone_frr() {
+  
+  local frr_config=$(mktemp -d -t frr-XXXXXXXXXX)
+  local -n vrfs=$1
+  generate_frr_config ${frr_config}/frr.conf vrfs
+  
+  cp "${FRR_TMP_DIR}"/frr-k8s/hack/demo/frr/daemons $frr_config
+  chmod a+rw ${frr_config}/*
+
+  # cleanup
+  $CLI rm -f frr || true
+  $CLI network rm -f ostestbm_net || true
+  
+  # create a $CLI network attached to the cluster network by attaching to the existing ostestbm bridge
+  $CLI network create --driver bridge --ipam-driver=none --opt com.docker.network.bridge.name=ostestbm ostestbm_net
+
+  $CLI run -d --rm --privileged --ulimit core=-1 --network ostestbm_net --name frr --volume "$frr_config":/etc/frr quay.io/frrouting/frr:9.1.0
   # ipv4 forwarding is enabled by default, we only need to turn on ipv6 forwarding
-  podman exec frr sysctl -w net.ipv6.conf.all.forwarding=1
+  $CLI exec frr sysctl -w net.ipv6.conf.all.forwarding=1
+  
+  # attach the frr container to the ostestbm bridge, so it can talk with the OCP nodes.
+  # use primary network gateway as the default route
+  $CLI exec frr ip address add dev eth0 192.168.111.3/24
+  $CLI exec frr ip route add default dev eth0 via 192.168.111.1
+  $CLI exec frr ip -6 address add dev eth0 fd2e:6f44:5dd8:c956::3/64
+  $CLI exec frr ip -6 route add default dev eth0 via fd2e:6f44:5dd8:c956::1
+
+  # eth1 attaches to agnhost_net
+  $CLI network connect agnhost_net frr
+  $CLI exec frr ip address add dev eth1 172.20.0.2/16
+  $CLI exec frr ip -6 address add dev eth1 2001:db8:2::2/64
+  
+  # setup VRFs for extra networks
+  seq=2 
+  for vrf in "${!vrfs[@]}"; do
+    [ "default" = "$vrf" ] && continue
+
+    # cleanup
+    $CLI network rm -f ${vrf}_net || true
+    
+    # create VRF
+    $CLI exec frr ip link add $vrf type vrf table $seq
+    $CLI exec frr ip link set dev $vrf up
+    $CLI exec frr ip route add table $seq unreachable default metric 4278198272
+    $CLI exec frr ip -6 route add table $seq unreachable default dev lo metric 4278198272
+
+    # create a $CLI network attached to the extra network
+    $CLI network create --driver bridge --ipam-driver=none --opt com.docker.network.bridge.name=${vrf} ${vrf}_net
+
+    # attach the frr container to the extra network and add to VRF
+    local subnet_v4_var=${vrf^^}_NETWORK_SUBNET_V4
+    local subnet_v6_var=${vrf^^}_NETWORK_SUBNET_V6
+    local ip=${!subnet_v4_var/\.0\//.3\/}
+    local ip6=${!subnet_v6_var/::\//::3\/}
+    $CLI network connect ${vrf}_net frr
+    $CLI exec frr ip link set dev eth$seq master $vrf
+    $CLI exec frr ip address add dev eth$seq $ip
+    $CLI exec frr ip -6 address add dev eth$seq $ip6
+
+    ((seq+=1))
+
+    # attach the frr container to the corresponding agnhost network and add to VRF
+    $CLI network connect agnhost_${vrf}_net frr
+    $CLI exec frr ip link set dev eth$seq master $vrf
+    $CLI exec frr ip address add dev eth$seq 172.20.0.2/16
+    $CLI exec frr ip -6 address add dev eth$seq 2001:db8:2::2/64
```

Comment:
Why use the same IP addresses for the interfaces in different VRFs? I know it works. But is there any particular reason?